File: frontend/src/app/usecase/workout/DeleteWorkoutLogUseCase.ts
import DeleteWorkoutLog from '../../port/in/workout/DeleteWorkoutLog';
import WorkoutLogClient from '../../port/out/WorkoutLogClient';

export default class DeleteWorkoutLogUseCase implements DeleteWorkoutLog {
  
  constructor(
    private readonly client: WorkoutLogClient 
  ){}
  
  delete(workoutId: string): Promise<void> {
    return this.client.delete(workoutId);
  }
  
}

File: backend/src/main/java/com/marmouset/workout/app/usecase/workout/DeleteWorkoutLogUseCase.java
package com.marmouset.workout.app.usecase.workout;

import com.marmouset.workout.app.port.in.workout.DeleteWorkoutLog;
import com.marmouset.workout.app.port.out.workout.WorkoutLogRepository;
import java.util.UUID;
import org.springframework.stereotype.Component;

@Component
class DeleteWorkoutLogUseCase implements DeleteWorkoutLog {

  private final WorkoutLogRepository repository;

  public DeleteWorkoutLogUseCase(WorkoutLogRepository repository) {
    this.repository = repository;
  }


  @Override
  public void delete(UUID uuid) {
    repository.delete(uuid);
  }

}
